src/UI/Window.cpp

```diff
@@ -766,3 +766,53 @@ void cWindow::SetProperty(short a_Property, short a_Value, cPlayer & a_Player)
 
 
 
+
+
+void cWindow::LoadRecipe(cPlayer & a_Player, UInt32 RecipeID)
+{
+	AString RecipeName = cRoot::Get()->GetRecipeMapper()->recipeIDToRecipeName(RecipeID, a_Player.GetClientHandle()->GetProtocolVersion());
+	cCraftingRecipes::cRecipe * Recipe = cRoot::Get()->GetCraftingRecipes()->findRecipeByRecipeName(RecipeName);
+
+	std::list<cItem> usedItems;
+	auto slotAreaCrafting = static_cast<cSlotAreaCrafting *>(m_SlotAreas[0]);
```

Comment:
This function definitely doesn't belong in `cWindow`. Probably belongs in `cSlotAreaCrafting`.